R/get_r_messages.R

```diff
@@ -37,16 +37,23 @@ get_r_messages <- function (dir, custom_translation_functions = NULL, is_base =
   #   <!-- mix and match those two types indefinitely -->
   #   <OP-RIGHT-PAREN>)</OP-RIGHT-PAREN>
   # </expr>
+  dots_funs <- domain_dots_funs(include_conditions)
+  fmt_funs <- domain_fmt_funs(include_conditions_f)
+
   singular_strings = rbind(
-    get_dots_strings(expr_data, DOMAIN_DOTS_FUNS, NON_DOTS_ARGS),
+    get_dots_strings(expr_data, dots_funs, NON_DOTS_ARGS),
     # treat gettextf separately since it takes a named argument, and we ignore ...
-    get_named_arg_strings(expr_data, 'gettextf', c(fmt = 1L), recursive = TRUE),
+    get_named_arg_strings(expr_data, fmt_funs, c(fmt = 1L), recursive = TRUE),
+    if (use_tr) get_named_arg_strings(expr_data, 'tr_', c(string = 1L), recursive = TRUE),
```

Comment:
an aside: `tr_` might be a good opportunity to build something I'd been kind of stuck on:

Since R doesn't have parser-level string concatenation (i.e. C's `'abc' 'def'`) making translation-ready strings in R will tend to produce lintr-violating wide lines (esp. at 80 but also at 120 characters)... the tooling for base has no shot here because it considers all the inputs as individual strings.

But with custom functions we could allow:

```
tr_(
  'abcd',
  'efg'
)
```

To wind up as `abcdefg` in the .pot file & also run `gettext()` on the combined string:

```
tr_ <- function(...)
  gettext(paste0(...))
```

I think it would require some effort to get it working but maybe worthwhile to prevent infinite scroll strings...

Reply:
Yeah, I like that idea. I'll change it to a dots function. 

In total we might want something like:

```R
tr_ <- function(...) {
  structure(enc2utf8(gettext(paste0(...), domain = "R-pkgdown")), class = "translated")
}
```

That ensures it's correctly flagged as UTF-8 (which `gettext()`) doesn't do by default, and flags it as a translated string which might be useful for other functions.

Reply:
This function could also error if given anything other than a literal string (i.e. `tr_(foo)` and tr_(foo())` should error because they're not translatable).

Reply:
that sounds basically right to me (requiring literal inputs), I do wonder if it will wind up being limiting. I have in mind the `N_()` construct in C which is needed to keep strings untranslated at compile time, then translated as a variable at run time.

https://github.com/wch/r-source/blob/430360aa394b02046dd7175c0fc55e347e947f80/src/main/errors.c#L1398-L1419

https://github.com/wch/r-source/blob/430360aa394b02046dd7175c0fc55e347e947f80/src/main/errors.c#L1461

Not sure if it's relevant, so probably it's OK to start being strict and allow the use cases to show themselves as a feature request.